File: api/src/main/java/com/sahilkumar/api/customer/Customer.java
package com.sahilkumar.api.customer;

import com.sahilkumar.api.auth.User;
import com.sahilkumar.api.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false, length = 120)
    private String name;

    @Column(nullable = false, length = 20)
    private String phone;

    @Column(length = 255)
    private String address;

    @Column(name = "custom_milk_rate", precision = 10, scale = 2)
    private BigDecimal customMilkRate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CustomerStatus status;
}

File: android/app/src/main/java/com/sahilkumar/dfms/model/Customer.kt
package com.sahilkumar.dfms.model

import java.math.BigDecimal

data class CustomerResponse(
    val id: String,
    val name: String,
    val phone: String,
    val address: String?,
    val customMilkRate: BigDecimal?,
    val status: CustomerStatus,
    val createdAt: String?,
    val userId: String?,
    val username: String?,
)

data class CustomerRequest(
    val name: String,
    val phone: String,
    val address: String?,
    val customMilkRate: BigDecimal?,
    val status: CustomerStatus,
)

data class CreateCustomerLoginRequest(
    val username: String,
    val password: String,
)
